### PR TITLE
Enable VOSK Transcription

### PR DESCRIPTION
**Pull request by janonym1**
this creates 2 (maybe) new ENV variables to allow for easier VOSK integration. Per default, it is assumed GCLOUD is always used and that makes it hard(er) to set up VOSK within docker-jitsi-meet

I still need to add the GCLOUD check and set for the ENV GOOGLE_APPLICATION_CREDENTIALS /config/key.json within the run script (jigasi/rootfs/etc/services.d/jigasi/run) but I am not sure how to best approach this. Maybe lets define an additional variable (ENABLE_GCLOUD_TRANSCRIPTION) and use that?

### CODE REVIEW

**Comment by janonym1** [comment]
I added a new var ENABLE_GCLOUD_TRANSCRIPTION but I am not sure, if my code at jigasi/rootfs/etc/services.d/jigasi/run seting the GCLOUD ENV for the key.json is working as intended

**Comment by janonym1** [comment]
I also am not sure, if it would be better/nicer if we can chain the check for the GCLOUD creds in the 10-config:

I wanted to OR together the following checks: 
`if [[ ($ENABLE_TRANSCRIPTIONS -eq 1 || $ENABLE_TRANSCRIPTIONS == "true") && ($ENABLE_VOSK -eq 0 || $ENABLE_VOSK == "false") && ($ENABLE_GCLOUD_TRANSCRIPTION -eq 1 || $ENABLE_GCLOUD_TRANSCRIPTION == "true")  ]]` but that isnt working as I imagined it.

File: jigasi.yml
Comment (saghul) [inline jigasi.yml:16]: For symmetry, this one should be called `ENABLE_VOSK_TRANSCRIPTION`

File: jigasi/rootfs/etc/cont-init.d/10-config
Comment (janonym1) [reply jigasi/rootfs/etc/cont-init.d/10-config]: I assume you mean the () here?

File: jigasi/rootfs/etc/services.d/jigasi/run
Comment (janonym1) [reply jigasi/rootfs/etc/services.d/jigasi/run:2]: then would it be fine, if I check if the key.json exists (and if yes import the ENV)?
```
if [[ -f /config/key.json ]]; then
    ENV GOOGLE_APPLICATION_CREDENTIALS /config/key.json
fi
```

Comment (janonym1) [reply jigasi/rootfs/etc/services.d/jigasi/run]: makes sense

**Review by janonym1 — COMMENTED** [review]
updated the requested changed

File: jigasi/rootfs/etc/cont-init.d/10-config
Comment (devByGelu) [inline jigasi/rootfs/etc/cont-init.d/10-config:29]: should this be `$ENABLE_VOSK_TRANSCRIPTION`?

**Comment by debendraoli** [comment]
Is there anything left to do? I would be happy to help.

It would be very useful when this gets merged. Thanks

**Comment by janonym1** [comment]
> Is there anything left to do? I would be happy to help.
> 
> It would be very useful when this gets merged. Thanks

I updated the requested changes (e.g. the variablenaming ENABLE_VOSK_TRANSCRIPTION) but I need to test it out on my testsystem first, which I unfortunately didnt get around to yet. If I botched up something, I also break the GCLOUD transcription ability, which more users seem to use than VOSK :)

Also I am not sure about the logic, that checks if transcriptions are on (but not gcloud) which I will get around to next week hopefully

**Comment by charles-zablit** [comment]
Hi, is there any update on this?

**Review by HarHarLinks — CHANGES_REQUESTED** [review]
it seems to me one variable was missed when renaming

**Comment by rouaidakacem** [comment]
@janonym1 is there an update on this ? Would love to help if necessary.

**Comment by HarHarLinks** [comment]
@debendraoli @charles-zablit @rouaidakacem and everyone else asking to help: could you test this PR and report if it works, and in case you have to make any changes, which ones? Because even with my proposed change, I can't get this working.

File: jigasi/rootfs/defaults/sip-communicator.properties
Comment (rouaidakacem) [inline jigasi/rootfs/defaults/sip-communicator.properties:13]: I think this needs a default

Comment (janonym1) [reply jigasi/rootfs/defaults/sip-communicator.properties:13]: > 

Maybe my jitsi setup deviated too far, since I previously scripted together the VOSK support by hand. I will try a fresh and clean install with the commit and see, what doesn't work. I may have forgotten a var or default, like rouaidakacem mentioned

Comment (rouaidakacem) [reply jigasi/rootfs/defaults/sip-communicator.properties:13]: Also I do not see the use of the varaible ENABLE_GCLOUD_TRANSCRIPTION it is defined in sip-communicator.properties but it is not used.

**Comment by saiflakhani** [comment]
Been patiently waiting for this, hoping to see this in a release soon! Would love to help/test if there's an update on this

**Comment by saghul** [comment]
There is currently a conflict that needs to be fixed.

**Comment by surajprakashchoudhary** [comment]
I am using docker-jitsi-meet-stable-8922 and I made the changes mentioned above by @janonym1. 
Steps followed by me:
- run command to install docker vosk : docker run -d -p 2700:2700 alphacep/kaldi-en:latest
- make changes in config.js.
Screenshot from 2023-12-04 16-44-52

- make changes in .env file
Screenshot from 2023-12-04 16-47-52

- make changes in Dockerfile at path  /root/docker-jitsi-meet-stable-8922/jigasi/Dockerfile
Screenshot from 2023-12-04 16-49-55

- make changes in sip-communicator.properties at path : /docker-jitsi-meet-stable-8922/jigasi/rootfs/defaults# nano sip-communicator.properties 
1. added variables .Env.VOSK_SERVER:
Screenshot from 2023-12-04 16-58-46
2. use variable .Env.VOSK_SERVER:
Screenshot from 2023-12-04 16-57-53

- make changes in /docker-jitsi-meet-stable-8922/jigasi/rootfs/etc/cont-init.d# nano 10-config
Screenshot from 2023-12-04 17-09-48

- make changes in run file : /docker-jitsi-meet-stable-8922/jigasi/rootfs/etc/services.d/jigasi# nano run
Screenshot from 2023-12-04 17-12-15

- after that I bulid the jigasi again using command : make build_jigasi from the root//docker-jitsi-meet-stable-8922 folder.

but still I am not able to see  start translation button in more options. please guide me where am I doing wrong 

Screenshot from 2023-12-04 17-28-40

**Comment by loelkes** [comment]
Hi @saghul What needs to be done here? I would like to help. It looks like the changes you requested have not been reviewed by you. Or have I missed something?

How can I/we help to get this merged?

**Comment by saghul** [comment]
Yep, conflicts are yet to be solved.

**Comment by loelkes** [comment]
@saghul I understand that. But from what I can see, the changes you requested have been made but not reviewed. Can you review them and point out the remaining conflicts?

I don't have write access here, so all I could do was create a new branch/PR to address the conflicts there.

**Comment by saghul** [comment]
Please do that. I can't merge the PR as is, even if the feedback was addressed, because there are conflicts.

**Comment by janonym1** [comment]
Sorry, I must have overlooked the activity here. I solved the conflicts but I think I have a bug somehwere in the logic. Let me look for my hardcoded values (since it worked for me before) and crosscheck with the effective values here

File: jigasi.yml
Comment (saghul) [inline jigasi.yml:15]: This change is not backwards compatible. I'd say we should either repurpose `ENABLE_TRANSCRIPTIONS` to take a boolean (default to gcloud) or the type of transcription.

Alternatively some `TRANSCRIPTION_BACKEND_TYPE` which one could set to `google` or `vosk`

File: jigasi/rootfs/etc/services.d/jigasi/run
Comment (saghul) [inline jigasi/rootfs/etc/services.d/jigasi/run:2]: Also check the transcriber type I think, simpler for testing switching between them.

File: jigasi/rootfs/defaults/sip-communicator.properties
Comment (saghul) [inline jigasi/rootfs/defaults/sip-communicator.properties:13]: This is unnecessary if you are not going to apply any transformation, you can use `.Env.VOSK_SERVER` directly.

**Review by saghul — CHANGES_REQUESTED** [review]

**Comment by zobadaniel** [comment]
I have a simlar patch, so I'd rather 👍 this than to post my own. I am also willing to help in order to get this merged.

**Comment by janonym1** [comment]
I forgot to add, that you have to have a working SIP setup (server+account) and config for VOSK to work. 
Should I just assume it is setup or make a check (e.g. if  JIGASI_SIP_URI is set)?

I also do not seem to get around the GCLOUD restriction; it needs to have a jigasi/config/key.json with **valid** credentials even when GCLOUD is deactivated. I am not sure how to get around this, maybe mock the credentials?

**Comment by damencho** [comment]
> I forgot to add, that you have to have a working SIP setup (server+account) and config for VOSK to work.
> Should I just assume it is setup or make a check (e.g. if JIGASI_SIP_URI is set)?

This is not a jigasi issue, this is configuration issue and should be fixed, maybe in this PR?

**Comment by saghul** [comment]
I think I fixed that already. If you set ENABLE_TRANSCRIPTIONS the sip config is skipped now.

**Comment by aslam-t** [comment]
Hey fellows, can we please prioritize this somehow? getting into compliance issues for some users.

**Comment by damencho** [comment]
Can you resolve the conflicts, please?

**Comment by damencho** [comment]
@saghul is there anything missing here?

**Comment by aslam-t** [comment]
> Please take a look at commit [M4GNV5@d44daf2](https://github.com/M4GNV5/docker-jitsi-meet/commit/d44daf2b7bc422ea0449961123178e2296834b61) which should address the open issues from @saghul - I am interested in getting this merged as well, thus I asked @M4GNV5 for this contribution.

Hey @damencho, these commits have no conflict with `jitsi/docker-jitsi-meet` `master` branch. Can your team please check/review?.

Also, I am willing to test this work, if there's a container image/tag for this. I am trying to create my own image, since its a good amount of learning curve for me, it will take time. @zobadaniel if it is possible for you to provide an image out of [M4GNV5@d44daf2](https://github.com/M4GNV5/docker-jitsi-meet/commit/d44daf2b7bc422ea0449961123178e2296834b61) branch, it will accelerate things.

Thanks

**Comment by damencho** [comment]
> > Please take a look at commit [M4GNV5@d44daf2](https://github.com/M4GNV5/docker-jitsi-meet/commit/d44daf2b7bc422ea0449961123178e2296834b61) which should address the open issues from @saghul - I am interested in getting this merged as well, thus I asked @M4GNV5 for this contribution.
> 
> Hey @damencho, these commits have no conflict with `jitsi/docker-jitsi-meet` `master` branch. Can your team please check/review?.
> 
> Also, I am willing to test this work **today**, if there's a container image/tag for this. I am trying to create my own image, since its a good amount of learning curve for me, it will take time. @zobadaniel if it is possible for you to provide an image out of [M4GNV5@d44daf2](https://github.com/M4GNV5/docker-jitsi-meet/commit/d44daf2b7bc422ea0449961123178e2296834b61) branch, it will accelerate things.
> 
> Thanks

image

**Comment by zobadaniel** [comment]
Dear @janonym1 can you please update this PR with https://github.com/M4GNV5/docker-jitsi-meet/commit/d44daf2b7bc422ea0449961123178e2296834b61 so that the requested changes are visible here?

**Comment by zobadaniel** [comment]
@aslam-t I have pushed my latest set of jitsi containers to docker, please look here: https://hub.docker.com/repositories/zalozbadev and pick the tag "stable-9584-1_custom-4" for each of the jitsi containers. It includes this PR and #1737 and works for me. I cannot update this PR myself, so the requested changes aren't directly visible. #1737 will be updated with the necessary changes soon, we need to sort out some merge issues there first.

**Comment by Marwane2185** [comment]
Hello @zobadaniel , I have tested your docker image for jigasi and did not work!
Also @janonym1  could please merge it to get changes in release.
I want to configure Jigasi with Vosk but did not work yet.

Thanks

**Comment by Marwane2185** [comment]
> stable-9584-1_custom-4

@zobadaniel , still getting  Transcriptions: One or more google cloud environment variables are undefined

**Comment by aaronkvanmeerten** [comment]
Hi y'all: sorry for the likely disruptive changes, but I wanted to update everybody on this thread about our plan to move some bits around in order to make the transcriber a separate parallel component.  This includes a new transcriber.yaml docker-compose file, and also will split the sip-communicator.properties file into pieces based on the jigasi mode.

All this to say I hope it makes the sort of proposed changes above easier to reason about, but will mean further updates on this PR.

I intend to get some progress on that in the next days, so I will link my PR when it's in any shape to discuss.

**Comment by aaronkvanmeerten** [comment]
My new PR splits out the config into different files based on jigasi role, and adds a new docker-compose file:
https://github.com/jitsi/docker-jitsi-meet/pull/1881
It is currently untested and I hope to get more time this week to move it further.

**Comment by Marwane2185** [comment]
Hello @aaronkvanmeerten , Thanks for update.
I'm wondering if it is possible to get transcript whenever a new user connect to conference (to provide him summary of what has been discussed previsouly). This will be very helpful.

Thanks

**Comment by aaronkvanmeerten** [comment]
> Hello @aaronkvanmeerten , Thanks for update.
> 
> I'm wondering if it is possible to get transcript whenever a new user connect to conference (to provide him summary of what has been discussed previsouly). This will be very helpful.
> 
> 
> 
> Thanks

This sounds like a late arrival summaries to me. I believe that might be possible, depending on how it is stored. This would be a customization from the iframe API or via custom client code however, it's not something we support out of the box.

**Comment by Ali-Aljufairi** [comment]
I have question is it possible to transcibe multiple lanaguages at the same time using gclound or vsok ?

**Comment by aaronkvanmeerten** [comment]
> I have question is it possible to transcibe multiple lanaguages at the same time using gclound or vsok ?

I feel this thread is getting out of context.  Please direct such questions to https://community.jitsi.org/ 
TL;DR; not so far as I know.

**Comment by Ali-Aljufairi** [comment]
> > I have question is it possible to transcibe multiple lanaguages at the same time using gclound or vsok ?
> 
> I feel this thread is getting out of context. Please direct such questions to https://community.jitsi.org/ TL;DR; not so far as I know.

This Pr is like 2 year old so its normall thanks for guidense I will ask there

**Comment by oerodriguezn** [comment]
> > Hello @aaronkvanmeerten , Thanks for update.
> > I'm wondering if it is possible to get transcript whenever a new user connect to conference (to provide him summary of what has been discussed previsouly). This will be very helpful.
> > Thanks
> 
> This sounds like a late arrival summaries to me. I believe that might be possible, depending on how it is stored. This would be a customization from the iframe API or via custom client code however, it's not something we support out of the box.

Hi, great job. I believe your changes are already in master branch, Do you have any estimated release date? Would you include this issue about being able to use, for example Vosk, instead of Google, without the need of configure google env vars?

Do you have any documentation about the features/bug fixes in this version? I'm about to add transcription to my JITSI, but maybe it is worth to wait for your release.

**Comment by oerodriguezn** [comment]
> Hi y'all: sorry for the likely disruptive changes, but I wanted to update everybody on this thread about our plan to move some bits around in order to make the transcriber a separate parallel component. This includes a new transcriber.yaml docker-compose file, and also will split the sip-communicator.properties file into pieces based on the jigasi mode.
> 
> All this to say I hope it makes the sort of proposed changes above easier to reason about, but will mean further updates on this PR.
> 
> I intend to get some progress on that in the next days, so I will link my PR when it's in any shape to discuss.

Hi @aaronkvanmeerten  , great job. I believe your changes are already in master branch, Do you have any estimated release date? Would you include this issue about being able to use, for example Vosk, instead of Google, without the need of configure google env vars?

Do you have any documentation about the features/bug fixes in this version? I'm about to add transcription to my JITSI, but maybe it is worth to wait for your release.

**Comment by aaronkvanmeerten** [comment]
> > [snip]
> > I intend to get some progress on that in the next days, so I will link my PR when it's in any shape to discuss.
> 
> Hi @aaronkvanmeerten , great job. I believe your changes are already in master branch, Do you have any estimated release date? Would you include this issue about being able to use, for example Vosk, instead of Google, without the need of configure google env vars?
> 
> Do you have any documentation about the features/bug fixes in this version? I'm about to add transcription to my JITSI, but maybe it is worth to wait for your release.

Here's one more PR that needs to be merged to fully support Vosk.
https://github.com/jitsi/docker-jitsi-meet/pull/1915

Once merged/marked stable then you'd just set:

JIGASI_TRANSCRIBER_CUSTOM_SERVICE=org.jitsi.jigasi.transcription.VoskTranscriptionService
JIGASI_TRANSCRIBER_VOSK_URL=wss://your-server.yourdomain.tld/path
 
Until the time it's merged, you can provide a custom jigasi config in /config/custom-sip-communicator.properties with the appropriate snippet it will be appended to the bottom of the file:
`org.jitsi.jigasi.transcription.vosk.websocket_url=wss://your-url.domain.tld/path`

**Comment by damencho** [comment]
Vosk config has been added recently.

**Comment by celevra** [comment]
the vars are working and so vosk but it still tries to contact google

**Comment by aaronkvanmeerten** [comment]
Can you show any logs?  I haven't tested specifically for vosk only as I don't have an endpoint for taht one.  I'm curious to see your jigasi configuration as well as any errors in startup?
specifically this variable SHOULD be setting properly the default transcriber: `JIGASI_TRANSCRIBER_CUSTOM_SERVICE`
should be setting `org.jitsi.jigasi.transcription.customService=` in the sip-communicator.properties file

**Comment by celevra** [comment]
```
jitsi-transcriber-1  | 2024-10-23 15:46:57.888 SEVERE: [163] VoskTranscriptionService$VoskWebsocketStreamingSession.onError#306: Error while streaming audio data to transcription service
jitsi-transcriber-1  | com.google.cloud.translate.TranslateException: Method doesn't allow unregistered callers (callers without established identity). Please use API Key or other form of API consumer identity to call this API.
jitsi-transcriber-1  |  at com.google.cloud.translate.spi.v2.HttpTranslateRpc.translate(HttpTranslateRpc.java:57)
jitsi-transcriber-1  |  at com.google.cloud.translate.spi.v2.HttpTranslateRpc.translate(HttpTranslateRpc.java:126)
jitsi-transcriber-1  |  at com.google.cloud.translate.TranslateImpl$4.call(TranslateImpl.java:124)
jitsi-transcriber-1  |  at com.google.cloud.translate.TranslateImpl$4.call(TranslateImpl.java:121)
jitsi-transcriber-1  |  at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:103)
jitsi-transcriber-1  |  at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
jitsi-transcriber-1  |  at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
jitsi-transcriber-1  |  at com.google.cloud.translate.TranslateImpl.translate(TranslateImpl.java:120)
jitsi-transcriber-1  |  at com.google.cloud.translate.TranslateImpl.translate(TranslateImpl.java:138)
jitsi-transcriber-1  |  at org.jitsi.jigasi.transcription.GoogleCloudTranslationService.translate(GoogleCloudTranslationService.java:60)
jitsi-transcriber-1  |  at org.jitsi.jigasi.transcription.TranslationManager.getTranslations(TranslationManager.java:149)
jitsi-transcriber-1  |  at org.jitsi.jigasi.transcription.TranslationManager.notify(TranslationManager.java:177)
jitsi-transcriber-1  |  at org.jitsi.jigasi.transcription.Transcriber.notify(Transcriber.java:871)
jitsi-transcriber-1  |  at org.jitsi.jigasi.transcription.Participant.notify(Participant.java:585)
jitsi-transcriber-1  |  at org.jitsi.jigasi.transcription.VoskTranscriptionService$VoskWebsocketStreamingSession.onMessage(VoskTranscriptionService.java:283)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.messages.StringMessageSink.accept(StringMessageSink.java:54)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.common.JettyWebSocketFrameHandler.acceptMessage(JettyWebSocketFrameHandler.java:351)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.common.JettyWebSocketFrameHandler.onTextFrame(JettyWebSocketFrameHandler.java:437)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.common.JettyWebSocketFrameHandler.onFrame(JettyWebSocketFrameHandler.java:241)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.WebSocketCoreSession$IncomingAdaptor.lambda$onFrame$1(WebSocketCoreSession.java:671)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.WebSocketCoreSession.handle(WebSocketCoreSession.java:118)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.WebSocketCoreSession$IncomingAdaptor.onFrame(WebSocketCoreSession.java:671)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.ExtensionStack.onFrame(ExtensionStack.java:120)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.WebSocketCoreSession.onFrame(WebSocketCoreSession.java:481)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.onFrame(WebSocketConnection.java:271)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.fillAndParse(WebSocketConnection.java:464)
jitsi-transcriber-1  |  at org.eclipse.jetty.websocket.core.internal.WebSocketConnection.onFillable(WebSocketConnection.java:349)
jitsi-transcriber-1  |  at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
jitsi-transcriber-1  |  at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
jitsi-transcriber-1  |  at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:421)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:390)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:277)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:199)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:969)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1194)
jitsi-transcriber-1  |  at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1149)
jitsi-transcriber-1  |  at java.base/java.lang.Thread.run(Thread.java:840)
jitsi-transcriber-1  | Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 403 Forbidden
jitsi-transcriber-1  | GET https://translation.googleapis.com/language/translate/v2?model=nmt&q=bin%20an&source=de&target=en
jitsi-transcriber-1  | {
jitsi-transcriber-1  |   "code" : 403,
jitsi-transcriber-1  |   "errors" : [ {
jitsi-transcriber-1  |     "domain" : "global",
jitsi-transcriber-1  |     "message" : "Method doesn't allow unregistered callers (callers without established identity). Please use API Key or other form of API consumer identity to call this API.",
jitsi-transcriber-1  |     "reason" : "forbidden"
jitsi-transcriber-1  |   } ],
jitsi-transcriber-1  |   "message" : "Method doesn't allow unregistered callers (callers without established identity). Please use API Key or other form of API consumer identity to call this API.",
jitsi-transcriber-1  |   "status" : "PERMISSION_DENIED"
jitsi-transcriber-1  | }
jitsi-transcriber-1  |  at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:146)
jitsi-transcriber-1  |  at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:118)
jitsi-transcriber-1  |  at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:37)
jitsi-transcriber-1  |  at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:428)
jitsi-transcriber-1  |  at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1111)
jitsi-transcriber-1  |  at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:514)
jitsi-transcriber-1  |  at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:455)
jitsi-transcriber-1  |  at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:565)
jitsi-transcriber-1  |  at com.google.cloud.translate.spi.v2.HttpTranslateRpc.translate(HttpTranslateRpc.java:112)
jitsi-transcriber-1  |  ... 37 more
```
this is my .env config

```
# Enable Jigasi transcription
ENABLE_TRANSCRIPTIONS=1
# Jigasi will post an url to the chat with transcription file [default: false]
JIGASI_TRANSCRIBER_ADVERTISE_URL=true

JIGASI_TRANSCRIBER_CUSTOM_SERVICE=org.jitsi.jigasi.transcription.VoskTranscriptionService
JIGASI_TRANSCRIBER_VOSK_URL=ws://jitsi-vosk:2700

# Credentials for connect to Cloud Google API from Jigasi
# Please read https://cloud.google.com/text-to-speech/docs/quickstart-protocol
# section "Before you begin" paragraph 1 to 5
# Copy the values from the json to the related env vars
#GC_PROJECT_ID=
#GC_PRIVATE_KEY_ID=
#GC_PRIVATE_KEY=
#GC_CLIENT_EMAIL=
#GC_CLIENT_ID=
#GC_CLIENT_CERT_URL=

# Enable recording
ENABLE_RECORDING=1

```

**Comment by aaronkvanmeerten** [comment]
It appears we need a new flag: `JIGASI_TRANSCRIBER_ENABLE_TRANSLATION`

https://github.com/jitsi/docker-jitsi-meet/pull/1953

Once this is merged and pushed into stable you can set that:
`JIGASI_TRANSCRIBER_ENABLE_TRANSLATION=0`

Until the time it's merged, you can provide a custom jigasi config in /config/custom-sip-communicator.properties with the appropriate snippet it will be appended to the bottom of the file:
org.jitsi.jigasi.transcription.ENABLE_TRANSLATION=false

**Pull request closed without merging**
